ui/src/features/Editor/useCanvasCopyPaste.ts

```diff
@@ -138,17 +125,53 @@ export default ({
       });
 
       return reBatchedNodes;
+    },
+    [handleWorkflowUpdate, newEdgeCreation],
+  );
+
+  const handleCopy = useCallback(async () => {
+    const selected: { nodeIds: string[]; edges: Edge[] } | undefined = {
+      nodeIds: nodes.filter((n) => n.selected).map((n) => n.id),
+      edges: edges.filter((e) => e.selected),
+    };
+
+    const selectedNodes = nodes.filter((n) => selected.nodeIds.includes(n.id));
+    if (selectedNodes.some((n) => n.type === "reader")) return;
+
+    if (selected.nodeIds.length === 0 && selected.edges.length === 0) return;
+    const copiedNodes = nodes.filter((n) => selected.nodeIds.includes(n.id));
```

Comment:
This is the same as selectedNodes, which CAN be the same as selected.nodes if you do the above suggestion ;)

Reply:
Also they aren't copied yet here